eureka2-core/src/main/java/com/netflix/eureka2/connection/RetryableConnectionFactory.java

```diff
@@ -124,5 +129,23 @@ public void call(CHANNEL channel) {
         );
     }
 
-    protected abstract Observable<Void> executeOnChannel(CHANNEL channel, OP op);
```

Comment:
I have added in my branch 
```java
protected abstract Observable<INPUT> connectToChannelInput(CHANNEL channel);
```
How can we expose channel input with this implementation?